### PR TITLE
✨ handling e2e testing with dockertest

### PR DESCRIPTION
**Pull request by ludusrusso**
Add testing of database.

Some basic tests, but I prefer to have some :D

Related to #36

### CODE REVIEW

File: generated/sqlc/db_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		Repository: "postgres",
+		Tag:        "13-alpine",
+		Env: []string{
+			"POSTGRES_USER=test",
+			"POSTGRES_PASSWORD=test",
+			"listen_addresses = '*'",
+		},
+		Mounts: []string{
+			fmt.Sprintf("%v/../../db/schema.sql:/docker-entrypoint-initdb.d/schema.sql", cdir),
+		},
```
Comment (ludusrusso) [inline generated/sqlc/db_test.go:43]: I'd like to better handle this line but don't know how

Comment (davideimola) [reply generated/sqlc/db_test.go:43]: What about path.Join?

https://golang.org/pkg/path/#example_Join

**Comment by davideimola** [comment]
Probably for e2e I will suggest to use ginkgo, because executes the test concurrently.

**Comment by ludusrusso** [comment]
> Probably for e2e I will suggest to use ginkgo, because executes the test concurrently.

Also with standart test libs we can handle tests in parallel :D

File: generated/sqlc/db_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "postgres",
+		Tag:        "13-alpine",
+		Env: []string{
+			"POSTGRES_USER=test",
+			"POSTGRES_PASSWORD=test",
+			"listen_addresses = '*'",
+		},
+		Mounts: []string{
+			fmt.Sprintf("%v/../../db/schema.sql:/docker-entrypoint-initdb.d/schema.sql", cdir),
```
Comment (davideimola) [inline generated/sqlc/db_test.go:42]: I think you can compose the first path using the `path.Join`.

**Review by davideimola — COMMENTED** [review]
What do you think?

File: generated/sqlc/db_test.go (hunk 2 above)
Comment (ludusrusso) [reply generated/sqlc/db_test.go:42]: I prefer to leave this now and handle better later wiith docker comopse file

**Review by davideimola — APPROVED** [review]
LGTM

**Pull request merged**